api/types/types.proto

```diff
@@ -1615,6 +1615,48 @@ message RoleConditions {
         (gogoproto.jsontag) = "windows_desktop_labels,omitempty",
         (gogoproto.customtype) = "Labels"
     ];
+
+    // RequireSessionJoin specifies policies for required users to start a session.
+    repeated SessionRequirePolicy RequireSessionJoin = 19
+        [ (gogoproto.jsontag) = "require_session_join,omitempty" ];
+
+    // JoinSessions specifies policies to allow users to join other sessions.
+    repeated SessionJoinPolicy JoinSessions = 20
+        [ (gogoproto.jsontag) = "join_sessions,omitempty" ];
+}
+
+// SessionRequirePolicy a requirement policy that needs to be fulfilled to grant access.
+message SessionRequirePolicy {
+    // Name is the name of the policy.
+    string Name = 1 [ (gogoproto.jsontag) = "name" ];
+
+    // Filter is a predicate that determines what users count towards this policy.
+    string Filter = 2 [ (gogoproto.jsontag) = "filter" ];
+
+    // Kinds are the session kinds this policy applies to.
+    repeated string Kinds = 3 [ (gogoproto.jsontag) = "kinds" ];
+
+    // Count is the amount of people that need to be matched for this policy to be fulfilled.
+    int32 Count = 4 [ (gogoproto.jsontag) = "count" ];
+
+    // OnLeave is the behaviour that's used when the policy is no longer fulfilled
+    // for a live session.
+    string OnLeave = 5 [ (gogoproto.jsontag) = "on_leave" ];
+}
+
+// SessionJoinPolicy defines a policy that allows a user to join sessions.
+message SessionJoinPolicy {
+    // Name is the name of the policy.
+    string Name = 1 [ (gogoproto.jsontag) = "name" ];
+
+    // Roles is a list of roles that you can join the session of.
```

Comment:
Neither sound quite right, to be honest. I would probably word it as

`Roles is a list of roles that can join the session.`

Reply:
Ah, this feels inverted though. Because a join policy applies to the person joining the session. This is a list of roles of which the user this policy is attached as can join, provided the session host has those roles.